src/rules.c

```diff
@@ -133,6 +132,34 @@ SchemaRule *SchemaRule_Create(SchemaRuleArgs *args, IndexSpec *spec, QueryError
   return NULL;
 }
 
+void RSExpr_GetProperties(RSExpr *e, char ***props);
+
+void SchemaRule_FilterFields(SchemaRule *rule) {
```

Comment:
Can you add a documentation here explaining what the function does and why its needed?